lib/operations/command_v2.js

```diff
@@ -31,6 +36,32 @@ class CommandOperationV2 extends OperationBase {
     // TODO: consider making this a non-enumerable property
     this.server = server;
 
+    const options = this.options;
+    const serverWireVersion = maxWireVersion(server);
+    const inTransaction = this.session && this.session.inTransaction();
+
+    if (this.readConcern && commandSupportsReadConcern(cmd) && !inTransaction) {
+      Object.assign(cmd, { readConcern: this.readConcern });
+    }
+
+    if (serverWireVersion >= SUPPORTS_WRITE_CONCERN_AND_COLLATION) {
+      if (this.writeConcern && this.hasAspect(Aspect.WRITE_OPERATION)) {
+        Object.assign(cmd, { writeConcern: this.writeConcern });
+      }
+
+      if (options.collation && typeof options.collation === 'object') {
```

Comment:
might not be priority here, but IIRC we are supposed to error if there is a collation passed in and it is not supported. Should we be doing that here?

Reply:
added